packages/dds/tree/src/feature-libraries/sequence-field/invert.ts

```diff
@@ -147,14 +173,22 @@ function invertMark<TNodeChange>(
 						// The nodes could have been attached but were not because of the source.
 						return [];
 					}
-					return [
-						{
-							type: "ReturnFrom",
-							id: mark.id,
-							count: mark.count,
-							detachedBy: mark.revision ?? revision,
-						},
-					];
+					const invertedMark: ReturnFrom<TNodeChange> = {
+						type: "ReturnFrom",
+						id: mark.id,
+						count: mark.count,
+						detachedBy: mark.revision ?? revision,
+					};
+
+					const movedChanges = crossFieldManager.get(
+						CrossFieldTarget.Destination,
+						mark.revision ?? revision,
+						mark.id,
+					);
+					if (movedChanges !== undefined) {
+						invertedMark.changes = movedChanges;
+					}
```

Comment:
Aren't we going to re-assign this in `transferMovedChanges` anyway?

Reply:
If we process the MoveOut first and the MoveIn second, this code will transfer the changes on the first pass. We won't be invalidated if we don't later process another MoveOut for this field, so we may never do the `amendInvert`/`transferMovedChanges` pass.

Reply:
How does the `crossFieldManager` determine that we successfully handled the transfer as part of this first pass?

Reply:
When processing the MoveIn we call `CrossFieldManager.get` which registers this field as depedent on the move ID. When processing the MoveOut we call `CrossFieldManager.getOrCreate` which invalidates dependents. If we process the MoveOut first we won't have created the dependency yet, so invalidation will happen. If invalidation happens anyway, due to some other MoveID, we will end up reassigning the child changes, but that should be harmless.